### ISSUE TITLE
options.headers doesn't get considered for ajax requests

### ISSUE THREAD

**Issue by hippostar** [i18next/i18next#429]
I'm using 1.10.2 without jQuery, and I was using custom headers which are defined using options.headers. 

These options are not properly forwarded to f.ajax when using the built-in function. I also suppose they also aren't applied to the jQuery ajax call when using the jquery version but I haven't investigated on that.

The options object passed to f.ajax is initialized in _fetchOne and doesn't include options.headers:

```javascript
_fetchOne: function(lng, ns, options, done) {
            var url = applyReplacement(options.resGetPath, { lng: lng, ns: ns });
            f.ajax({
                url: url,
                cache: options.cache,
                success: function(data, status, xhr) {
                    f.log('loaded: ' + url);
                    done(null, data);
                },
                error : function(xhr, status, error) {
                    if ((status && status == 200) || (xhr && xhr.status && xhr.status == 200)) {
                        // file loaded but invalid json, stop waste time !
                        f.error('There is a typo in: ' + url);
                    } else if ((status && status == 404) || (xhr && xhr.status && xhr.status == 404)) {
                        f.log('Does not exist: ' + url);
                    } else {
                        var theStatus = status ? status : ((xhr && xhr.status) ? xhr.status : null);
                        f.log(theStatus + ' when loading ' + url);
                    }
                    
                    done(error, {});
                },
                dataType: "json",
                async : options.getAsync,
                timeout: options.ajaxTimeout
            });
        },
```

**Comment by hippostar** [comment]
Workaround for now is adding **headers: options.headers** after timeout.

**Comment by jamuhl** [comment]
a PR solving your issue would be very welcome and be merged to next v1.10.x. right now i started work on a complete rewrite for v2.0.0 so the time is to limited right now.

**Comment by hippostar** [comment]
#431
